### PR TITLE
Home work4

### PR DESCRIPTION
**Pull request by RedAggressor**

### CODE REVIEW

File: HomeWork 4/HomeWork 4/Program.cs [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.Write("Enter length of array what you want to create at 0 for 255: ");
+
+            Console.WriteLine();
+
+            if (!byte.TryParse(Console.ReadLine(), out byte lengthArray) || lengthArray < 0 || lengthArray > byte.MaxValue)
```
Comment (PeterChernykh) [inline HomeWork 4/HomeWork 4/Program.cs]: Why  lengthArray > byte.MaxValue? I am not sure that this is possible in this case

File: HomeWork 4/HomeWork 4/Program.cs [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+            {
+                Console.WriteLine($"Error!!! enter only numbers at 0 for {byte.MaxValue}!");
+
+                return;
+            }
```
Comment (PeterChernykh) [inline HomeWork 4/HomeWork 4/Program.cs]: Better to use do_while for endless cycle

File: HomeWork 4/HomeWork 4/Program.cs [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+                    array[i] = randomNumber;
+                }
+                else
+                {
+                    array[i] = randomNumber + 1;
+                }
+            }
+
+            return array;
+        }
```
Comment (PeterChernykh) [inline HomeWork 4/HomeWork 4/Program.cs]: This is overengineering, we can use one method for both purposes

File: HomeWork 4/HomeWork 4/Program.cs [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            }
+            else if (firstValue == secondValue)
+            {
+                Console.WriteLine("array with upper letter even numbers = array with upper letter odd numbers");
+            }
+            else if (firstValue < secondValue)
+            {
+                Console.WriteLine("array with upper letter odd numbers > array with upper letter even numbers");
+            }
+        }
```
Comment (PeterChernykh) [inline HomeWork 4/HomeWork 4/Program.cs]: Better to use switch_case

File: HomeWork 4/HomeWork 4/Program.cs [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+            char[] changeOddArray = ChangeNumberToChar(arrayOddNumbers);
+
+            ChangeOnUpperCaseSpecificChar(changeOddArray);
+
+            int countCharUpperOfOdd = CountOfUpperChar(changeOddArray);
+
+            Console.WriteLine("Displey Odd char array with Upper: ");
+
+            DisplayShowCharArray(changeOddArray);
```
Comment (PeterChernykh) [inline HomeWork 4/HomeWork 4/Program.cs]: These actions are quite similar extra method can be created for this purpose

File: HomeWork 4/HomeWork 4/Program.cs
Comment (PeterChernykh) [inline HomeWork 4/HomeWork 4/Program.cs:1]: Where is the StyleCop?

File: HomeWork 4/HomeWork 4/Program.cs (hunk 1 above)
Comment (RedAggressor) [reply HomeWork 4/HomeWork 4/Program.cs:11]: if i use int.MaxVAlue will error with memory owersize. That's way i this parametr

File: HomeWork 4/HomeWork 4/Program.cs (hunk 2 above)
Comment (RedAggressor) [reply HomeWork 4/HomeWork 4/Program.cs:16]: Thanks, It`s greate idea!))

File: HomeWork 4/HomeWork 4/Program.cs (hunk 3 above)
Comment (RedAggressor) [reply HomeWork 4/HomeWork 4/Program.cs:143]: I thought about it but decided to do separate what))

File: HomeWork 4/HomeWork 4.sln
Comment (PeterChernykh) [inline HomeWork 4/HomeWork 4.sln:1]: I see that .gitignore is not working in your proj, please double check whether .gitignore added correctly

**Pull request merged**